### DISCUSSION TITLE
Debugging, development and testing of eclipse-platform

### DISCUSSION THREAD

**Discussion by yangmeng66** [eclipse-platform/eclipse.platform.releng.aggregator#845] [Q&A] [answered]
Hello, ask for advice,
I am going to develop the debugging function of PLSQL based on eclipse-platform. The reference is JDT-DEBUG
But I don't know how to get the debugger for debugging and development? Do you need to build sdk every time? You also know that this build sdk takes a lot of time

**Comment by merks** [comment]
Just get it from here:

https://download.eclipse.org/eclipse/updates/4.27-I-builds/

Or use the Eclipse SDK as your IDE where everything you'd need is already installed in that IDE and with the IDE as the target platform (which is the default), generally everything you need is available for your plugin development:

https://download.eclipse.org/eclipse/downloads/drops4/S-4.27M1-202301041800/

**Reply by yangmeng66** [reply]
First of all, thank you for your patience!
Sorry, I am Chinese, and my English is not very good. I know how to get the source code of eclipse, but I don't know how to test the debugger program I developed myself

**Comment by merks** [comment]
One thing I would suggest is setting up a development environment with all the source code projects of the Eclipse SDK:

https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md#create-an-eclipse-development-environment

This is is useful because then you can see how everything works and can search all source code easily.  This is the environment that many of the platform developers use to develop the platform...

One of the things in this environment is this launch configuration:

![image](https://user-images.githubusercontent.com/208716/217202704-f32e185e-2b75-44a2-af1c-0d5732791ee7.png)

This lets you **debug** launch an Eclipse IDE that uses all the plugins in the workspace, so you can set breakpoints and see how things work.  This way you can see how JDT's debugger works so that you can do similar things to extend it...

You can do the same setup in your own Eclipse SDK installation containing your own projects.

It's also very helpful in general to invoke  "Add all plugins..." via the quick search action (magnifying glass):

![image](https://user-images.githubusercontent.com/208716/217203514-0247aa49-bbd2-4451-afcd-521c02ed2a97.png)

This ensures that Ctrl-Shift-T can find not only the classes in your workspace but also those available in the target platform, e.g., the debug framework classes that you want to use.

**Comment by merks** [comment]
One thing I would suggest is setting up a development environment with all the source code projects of the Eclipse SDK:

https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md#create-an-eclipse-development-environment

This is is useful because then you can see how everything works and can search all source code easily.  This is the environment that many of the platform developers use to develop the platform...

One of the things in this environment is this launch configuration:

![image](https://user-images.githubusercontent.com/208716/217202704-f32e185e-2b75-44a2-af1c-0d5732791ee7.png)

This lets you **debug** launch an Eclipse IDE that uses all the plugins in the workspace, so you can set breakpoints and see how things work.  This way you can see how JDT's debugger works so that you can do similar things to extend it...

You can do the same setup in your own Eclipse SDK installation containing your own projects.

It's also very helpful in general to invoke  "Add all plugins..." via the quick search action (magnifying glass):

![image](https://user-images.githubusercontent.com/208716/217203514-0247aa49-bbd2-4451-afcd-521c02ed2a97.png)

This ensures that Ctrl-Shift-T can find not only the classes in your workspace but also those available in the target platform, e.g., the debug framework classes that you want to use.

**Reply by yangmeng66** [reply]
Hello, I'm ready for the environment. The source code can be seen in the eclipse-altform debug, but the source code can't be seen in the JDT part, and the breakpoint can't be broken. Is there any good way to import the JDT source code into eclipse (https://github.com/eclipse-jdt)?

![image](https://user-images.githubusercontent.com/51857567/217245156-7688ddbd-9547-4099-a19d-99298e17e5cc.png)
![image](https://user-images.githubusercontent.com/51857567/217245175-2ef29cd5-4438-4f54-a60a-debd78758094.png)

**Reply by yangmeng66** [reply]
Is there any way to run this source code? ( https://github.com/eclipse-jdt/eclipse.jdt.debug )
![image](https://user-images.githubusercontent.com/51857567/217246462-0b40597a-7ff0-40be-9fd0-6de7622aa934.png)

**Comment by merks** [comment]
You don't need source projects to set breakpoints.

JDT also has an Oomph setup which should do a better job importing project such that there are no errors:

https://github.com/eclipse-jdt/.github/blob/main/CONTRIBUTING.md#create-an-eclipse-development-environment

**Reply by yangmeng66** [reply]
Hello, I have used oomph to import the JDT project. Next, I have two questions:
First, clicking the source code will report an error, and then this error is a constant reminder? What is the reason for this?
![image](https://user-images.githubusercontent.com/51857567/217267706-ad7f3b76-ec5c-4b75-acd5-adb8fc3c43aa.png)
note:  this is my eclipse IDE version
![image](https://user-images.githubusercontent.com/51857567/217268113-eac4f5f3-3b0a-4fdb-abc6-08eb39a1f5ed.png)


Second: How should this project be used? I want to see debug the code logic?

**Comment by merks** [comment]
I would expect the setup to install the 4.27 SDK product:

![image](https://user-images.githubusercontent.com/208716/217268939-4f1ea1e7-b663-4279-b93b-9ed6010ea065.png)

You appear to have installed into an older IDE so I would suggest installing into a newer IDE certainly 2022-12 4.26 at least...

**Reply by yangmeng66** [reply]
Thank you very much. It's OK. Can you guide me how to run it next? (Run this source code)

**Comment by merks** [comment]
I suggest Google "eclipse debug launch ide" and read some of that information.  You'll need to create a launch configuration like what I showed in https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/discussions/845#discussioncomment-4891501

**Reply by yangmeng66** [reply]
Coincidentally, hello, after I import the JDT code, the result of running now is the eclipse IDE. Is the JDT plug-in used in my source code or in the eclipse IDE itself?

**Comment by merks** [comment, marked as answer]
Your launch configuration's plug-ins tab determine what's in the application that you launch.  But default it's all the things in the workspace and all the things in the target platform (where the later by default is all the things in the host IDE); the plugins in the workspace generally override the ones in the target platform.

**Reply by yangmeng66** [reply]
Nice, I found it. Next, wish me good luck!
By the way, I can only find the documents about this（ https://www.eclipse.org/articles/Article-Debugger/how-to.html ）But this is 2004 and it is very old. Do you know what else you can refer to?

**Comment by merks** [comment]
The articles provided by @vogella are high quality, e.g., 

https://www.vogella.com/tutorials/EclipseDebugging/article.html

**Reply by yangmeng66** [reply]
Ok, thank you very much. I will work hard. Thank you for your reply and wish you a wonderful day
